### PR TITLE
Rename static Zen1 settings

### PR DESCRIPTION
**Pull request by DaveCTurner**
Renames the following settings to remove the mention of `zen` in their names:

- `discovery.zen.hosts_provider` -> `discovery.seed_providers`
- `discovery.zen.ping.unicast.concurrent_connects` -> `discovery.seed_resolver.max_concurrent_resolvers`
- `discovery.zen.ping.unicast.hosts.resolve_timeout` -> `discovery.seed_resolver.timeout`
- `discovery.zen.ping.unicast.hosts` -> `discovery.seed_hosts`

The old names are still supported, but deprecated, and we reject attempts to
use both old and new names at the same time.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
@elasticmachine run elasticsearch-ci/1

File: docs/reference/modules/network.asciidoc [hunk 1]
```diff
@@ -28,19 +28,18 @@ because `:` is a special character in YAML.
...
+or hostnames. If a hostname lookup resolves to multiple IP addresses then each
+IP address will be used for discovery.
 https://en.wikipedia.org/wiki/Round-robin_DNS[Round robin DNS] -- returning a
 different IP from a list on each lookup -- can be used for discovery; non-
-existent IP addresses will throw exceptions and cause another DNS lookup
-on the next round of pinging (subject to JVM DNS caching).
-+
-Defaults to `["127.0.0.1", "[::1]"]`.
+existent IP addresses will throw exceptions and cause another DNS lookup on the
+next round of pinging (subject to JVM DNS caching).  + Defaults to
```
Comment (ywelsch) [inline docs/reference/modules/network.asciidoc]: there is a + here

File: server/src/main/java/org/elasticsearch/discovery/DiscoveryModule.java [hunk 2]
```diff
@@ -90,15 +94,16 @@ public DiscoveryModule(Settings settings, ThreadPool threadPool, TransportServic
...
                 }
             });
             BiConsumer<DiscoveryNode, ClusterState> joinValidator = plugin.getJoinValidator();
             if (joinValidator != null) {
                 joinValidators.add(joinValidator);
             }
         }
-        List<String> hostsProviderNames = DISCOVERY_HOSTS_PROVIDER_SETTING.get(settings);
+
+        List<String> hostsProviderNames = getSeedProviderNames(settings);
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/discovery/DiscoveryModule.java]: rename this variable as well?

File: server/src/main/java/org/elasticsearch/discovery/UnicastConfiguredHostsResolver.java [hunk 3]
```diff
@@ -41,8 +41,6 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Consumer;
 
-import static org.elasticsearch.discovery.zen.UnicastZenPing.DISCOVERY_ZEN_PING_UNICAST_CONCURRENT_CONNECTS_SETTING;
-
 public class UnicastConfiguredHostsResolver extends AbstractLifecycleComponent implements ConfiguredHostsResolver {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/discovery/UnicastConfiguredHostsResolver.java]: we can rename these classes in a follow-up

File: server/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java [hunk 4]
```diff
@@ -96,10 +96,17 @@
...
-    public static final Setting<TimeValue> DISCOVERY_ZEN_PING_UNICAST_HOSTS_RESOLVE_TIMEOUT =
-        Setting.positiveTimeSetting("discovery.zen.ping.unicast.hosts.resolve_timeout", TimeValue.timeValueSeconds(5), Property.NodeScope);
+
+    public static final Setting<Integer> LEGACY_DISCOVERY_ZEN_PING_UNICAST_CONCURRENT_CONNECTS_SETTING =
+        Setting.intSetting("discovery.zen.ping.unicast.concurrent_connects", 10, 0, Property.NodeScope, Property.Deprecated);
+    public static final Setting<TimeValue> LEGACY_DISCOVERY_ZEN_PING_UNICAST_HOSTS_RESOLVE_TIMEOUT =
+        Setting.positiveTimeSetting("discovery.zen.ping.unicast.hosts.resolve_timeout", TimeValue.timeValueSeconds(5),
+            Property.NodeScope, Property.Deprecated);
+
+    public static final Setting<Integer> DISCOVERY_SEED_RESOLVER_MAX_CONCURRENT_RESOLVERS_SETTING =
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java]: a bit ugly that we have to define this new setting in this class that's going away. We can clean that up later though when removing Zen1

**Review by ywelsch — APPROVED** [review]

**Pull request merged**